src/Core/Grid/Query/OutstandingQueryBuilder.php

```diff
@@ -0,0 +1,218 @@
+    {
+        $qb = $this->getBaseQueryBuilder($searchCriteria)
+            ->addSelect('oi.id_order_invoice AS id_invoice, oi.date_add')
+            ->addSelect('CONCAT(LEFT(c.`firstname`, 1), \'. \' , c.`lastname`) AS customer')
+            ->addSelect('c.company, rl.name AS risk, r.color')
+            ->addSelect('c.outstanding_allow_amount')
+            ->addSelect('c.id_customer, o.id_order')
+            ->addSelect('cur.iso_code');
+        $this->applySorting($qb, $searchCriteria);
+        $this->criteriaApplicator->applyPagination($searchCriteria, $qb);
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        return $this->getBaseQueryBuilder($searchCriteria)->select('COUNT(oi.id_order_invoice)');
+    }
+
+    /**
+     * @param SearchCriteriaInterface $criteria
+     *
+     * @return QueryBuilder
+     */
+    private function getBaseQueryBuilder(SearchCriteriaInterface $criteria): QueryBuilder
+    {
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'order_invoice', 'oi')
+            ->leftJoin('oi', $this->dbPrefix . 'orders', 'o', 'oi.id_order = o.id_order')
+            ->leftJoin('o', $this->dbPrefix . 'customer', 'c', 'o.id_customer = c.id_customer')
+            ->leftJoin('c', $this->dbPrefix . 'risk', 'r', 'c.id_risk = r.id_risk')
+            ->leftJoin('r', $this->dbPrefix . 'risk_lang', 'rl', 'r.id_risk = rl.id_risk AND rl.id_lang = :context_lang_id')
+            ->leftJoin('o', $this->dbPrefix . 'currency', 'cur', 'o.id_currency = cur.id_currency')
+            ->andWhere('number > 0')
+            ->andWhere('o.id_shop IN (:context_shop_ids)')
+            ->setParameter('context_lang_id', $this->contextLangId, PDO::PARAM_INT)
+            ->setParameter('context_shop_ids', $this->contextShopIds, Connection::PARAM_INT_ARRAY);
+
+        $this->applyFilters($qb, $criteria->getFilters());
+
+        return $qb;
+    }
+
+    /**
+     * Apply filters for query builder.
+     *
+     * @param QueryBuilder $qb
+     * @param array $filters
+     */
+    private function applyFilters(QueryBuilder $qb, array $filters): void
+    {
+        $strictComparisonFilters = [
+            'id_invoice' => 'oi.id_order_invoice',
+            'risk' => 'r.id_risk',
+            'outstanding_allow_amount' => 'c.outstanding_allow_amount',
+        ];
+
+        $likeComparisonFilters = [
+            'customer' => 'CONCAT(LEFT(c.firstname, 1), \'. \' , c.lastname)',
+            'company' => 'c.company',
+        ];
+
+        $dateComparisonFilters = [
+            'date_add' => 'oi.date_add',
+        ];
+
+        foreach ($filters as $filterName => $filterValue) {
+            if (isset($strictComparisonFilters[$filterName])) {
+                $alias = $strictComparisonFilters[$filterName];
```

Comment:
👍

Reply:
What do you mean?

Reply:
I like how you dit it :) so it's just a "well done !"

Using `$strictComparisonFilters` and `$likeComparisonFilters` is a good way to organize this code.

Code review is also about saying when we think something is well crafted 😄